### PR TITLE
Improvements to the Token API

### PR DESCRIPTION
**Pull request by cutwater**
* Token API is moved from UI to v3 and now is served at ``<prefix>/v3/auth/token/``.
* Token API does not support ``GET`` method anymore,
  token is returned to client only once after creation.
* Add support of HTTP Basic authentication method to the Token API.

Issue: #187

### CODE REVIEW

**Comment by cutwater** [comment]
Note: Affected files were reformatted with `black`.

**Comment by alikins** [comment]
I wouldn't mind if the black/style changes were in a separate PR, just to keep the git attribution clean.
Would make this pr smaller and easier to review. But I'm not too concerned about it.

**Comment by alikins** [comment]
openapi.yaml needs updating.

File: galaxy_ng/app/api/v3/views/auth.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+from rest_framework.request import Request
+from rest_framework.authtoken.models import Token
+from rest_framework.response import Response
+from rest_framework import status as http_code
+
+from galaxy_ng.app.api import base as api_base
+from galaxy_ng.app.api.permissions import RestrictOnCloudDeployments
+
+
+class TokenView(api_base.APIView):
```
[the remark this answers is not in the document]
Comment (cutwater) [reply galaxy_ng/app/api/v3/views/auth.py:13]: `ObtainAuthToken` class handles username and password passed in request body. Not sure if that's desirable.

[the remark this answers is not in the document]
Comment (alikins) [reply galaxy_ng/app/api/v3/views/auth.py:13]: Doesn't matter much. just wanting to verify that there we aren't missing using a standard idiom. I don't think there is much going on in ObtainAuthToken() so the approach in the pr is fine.

**Pull request merged**